clang_format_init.sh

```diff
@@ -0,0 +1,11 @@
+#!/bin/bash
```

Comment:
I see this goes and gets the package needed to run `clang-format`, but do we also want to have a `Makefile` incantation to do the formatting, so we can run it ad-hoc without a commit?

Reply:
Yeah, so I added two incantations
1. clang-format-init: configure the hooksPath, install clang-format CLI if not existent locally
2. format: to format files ad-hoc without a commit

Reply:
Rather than creating a separate script, thinking we should just add this to `bootstrap.sh` with the rest of the building and formatting dependencies. That way a new dev can run `bootstrap.sh` and have everything they need.

My preference is to remove the pre-commit hook so commits are predictable and just add the `format` target to the `Makefile`